tests/unit_tests/gui/test_main_window.py

```diff
@@ -0,0 +1,318 @@
+        # equal to the number of realizations
+        realization_widget = run_dialog._tab_widget.currentWidget()
+        assert isinstance(realization_widget, RealizationWidget)
+        list_model = realization_widget._real_view.model()
+        assert list_model.rowCount() == simulation_panel.ert.getEnsembleSize()
+
+        qtbot.mouseClick(run_dialog.done_button, Qt.LeftButton)
+
+    QTimer.singleShot(1000, use_rundialog)
+    qtbot.mouseClick(start_simulation, Qt.LeftButton)
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_the_plot_window_contains_the_expected_elements(
+    esmda_has_run, opened_main_window, qtbot
+):
+    gui = opened_main_window
+
+    # Click on Create plot after esmda has run
+    plot_tool = gui.tools["Create plot"]
+    plot_tool.trigger()
+
+    # Then the plot window opens
+    qtbot.waitUntil(lambda: gui.findChild(PlotWindow) is not None)
+    plot_window = gui.findChild(PlotWindow)
+    assert isinstance(plot_window, PlotWindow)
+
+    case_selection = plot_window.findChild(CaseSelectionWidget)
+    data_types = plot_window.findChild(DataTypeKeysWidget)
+    assert isinstance(data_types, DataTypeKeysWidget)
+    combo_boxes: List[QComboBox] = case_selection.findChildren(
+        QComboBox
+    )  # type: ignore
+
+    # Assert that the Case selection widget contains the expected cases
+    case_names = []
+    assert len(combo_boxes) == 1
+    combo_box = combo_boxes[0]
+    for i in range(combo_box.count()):
+        data_names = []
+        combo_box.setCurrentIndex(i)
+        case_names.append(combo_box.currentText())
+    assert case_names == [
+        "default",
+        "default_0",
+        "default_1",
+        "default_2",
+        "default_3",
+    ]
+
+    data_names = []
+    data_keys = data_types.data_type_keys_widget
+    for i in range(data_keys.model().rowCount()):
+        index = data_keys.model().index(i, 0)
+        data_names.append(str(index.data(Qt.DisplayRole)))
+    assert data_names == [
+        "POLY_RES@0",
+        "COEFFS:COEFF_A",
+        "COEFFS:COEFF_B",
+        "COEFFS:COEFF_C",
+    ]
+
+    assert {
+        plot_window._central_tab.tabText(i)
+        for i in range(plot_window._central_tab.count())
+    } == {
+        "Cross case statistics",
+        "Distribution",
+        "Gaussian KDE",
+        "Ensemble",
+        "Histogram",
+        "Statistics",
+    }
+
+    # Add all the cases
+    for name in case_names:
+        combo_box = combo_boxes[-1]
+        index = combo_box.findData(name)
+        combo_box.setCurrentIndex(index)
+        qtbot.mouseClick(case_selection.findChild(QToolButton), Qt.LeftButton)
+        combo_boxes: List[QComboBox] = case_selection.findChildren(
+            QComboBox
+        )  # type: ignore
+    assert len(case_selection.findChildren(QComboBox)) == len(case_names) + 1
+
+    # Cycle through showing all the tabs and plot each data key
+    for tab in plot_window._plot_widgets:
```

Comment:
perhaps loop through the data_keys in the outer loop, and then select each tab that is visible for that key. Perhaps you want some assertion on which tabs are enabled based on the kind of data_key it is?